### PR TITLE
Refactor reusable code from controller-bootstrap

### PR DESCRIPTION
**Pull request by JamesJHPark**
Issue: [aws-controllers-k8s/community/issues/1358](https://github.com/aws-controllers-k8s/community/issues/1358)

Description of changes:

- This PR is to refactor reusable code from controller-bootstrap in aws-controllers-k8s/pkg repository.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,3 +1,20 @@
 module github.com/aws-controllers-k8s/pkg
 
 go 1.18
+
+require gopkg.in/src-d/go-git.v4 v4.13.1
```
Comment (vijtrip2) [inline go.mod:5]:
```suggestion
require (
    gopkg.in/src-d/go-git.v4 v4.13.1
)
```

File: util/common.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+// prefix if it's missing.
+func EnsureSemverPrefix(s string) string {
+	// trim all leading 'v' runes (characters)
+	s = strings.TrimLeft(s, "v")
+	return fmt.Sprintf("v%s", s)
+}
+
+// GetSDKVersion returns the github.com/aws/aws-sdk-go version to use
+// from the --aws-sdk-go-version flag.
+func GetSDKVersion(awsSDKGoVersion string) string {
```
Comment (vijtrip2) [inline util/common.go:161]: this function does nothing. remove it.

File: util/git.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package util
```
Comment (vijtrip2) [inline util/git.go:14]: consider move this to new `git` package, instead of `util`

File: util/common.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package util
```
Comment (vijtrip2) [inline util/common.go:14]: consider moving this to new `io` package instead of `util`

**Review by vijtrip2 — CHANGES_REQUESTED** [review]

**Review by vijtrip2 — APPROVED** [review]

**Review by RedbackThomson — APPROVED** [review]
Good start. Thanks

**Comment by jaypipes** [comment]
Hold off on merging this please. I have a few suggestions will put in review later.

**Review by jaypipes — CHANGES_REQUESTED** [review]
@JamesJHPark This isn't quite what I had in mind, actually... I was thinking that we would place this code into the `pkg/` directory of the *code-generator* repository and then import it from controller-bootstrap.

The `pkg/` source code repository is really meant for things that are common or generic across all of ACK, and in the future, even broader than ACK. I wanted to keep the dependencies for the `pkg/` source code repository as minimal as possible, and adding in dependencies on things like `go-git` I think should remain in the code-generator source code repository.

**Comment by JamesJHPark** [comment]
Closing as per discussion. Please refer to [PR](https://github.com/aws-controllers-k8s/code-generator/pull/355).

**Pull request closed without merging**